### ISSUE TITLE
Ability to have turrets lock on to asteroids for mining 

### ISSUE THREAD

**Issue by KyroFirehammer** [endless-sky/endless-sky#2886]
It would be nice to have the ability to have turrets target asteroids when trying to mine. This would be very useful in cases such as when you're starting out with the star barge or if uou capptured a kor mereti model 8 and are using it it mine (although that would be rather silly unless you are not piloting it)
I'm not sure if there is a reason this isn't possible or it is and could be a very nice addition for small ships with turrets that could have poor steering (such as the barge )

**Comment by tehhowch** [comment]
The player can't lock onto asteroids.
That's the basis of the argument right now, with the crux being that if you could lock onto an asteroid, you would not have to hunt for them, and you can simply spam autofire and "next target" to mine. Mining already pays 900 - 5000+ credits per ton of pure profit without needing to jump to a new system, so requiring the player to actually attempt to find a minable asteroid is how the profit is balanced.

To mine with turrets, set your turret preference to "focused" and proceed as though you had a fixed position weapon.

**Comment by Hadron1776** [comment]
> To mine with turrets, set your turret preference to "focused" and proceed as though you had a fixed position weapon.

That will only work as long as there are no enemies nearby.

**Comment by KyroFirehammer** [comment]
Or it could work if only escorts could lock on and move the ship similarly to how if you re locked on an enemy it auto adjusts your angling to be facing them, the escorts could "lock on" as already seen with the model 8. This could also be extremely useful if their was a hotkey to set escorts to mine asteroids if you are amassing a fleet and part of it you are waiting on to arrive at the system and they are several jumps away

**Comment by Hadron1776** [comment]
And then you get free auto mining which has basically the same problem as being able to target the asteroids yourself. Unless I'm missing something.

**Comment by tehhowch** [comment]
Escort mining is unlikely, for the exact reason I mentioned above: mining is very profitable. See also #2405

**Comment by endless-sky** [comment]
Mining profits are weighted based on the assumption that it's a sort of mini-game for players who are just starting out in a new ship to earn a profit in a way that's less boring than cargo but less dangerous than fighting pirates. Any change that made mining easier for large ships or fleets would require rebalancing it to be less lucrative, which would make it no longer attractive to players early on.

**Comment by ghoulavenger** [comment]
Considering how long it takes to pop an asteroid with basic lasers, and how difficult it is to differentiate between minable asteroids and non-minable asteroids, I already think it isn't very lucrative early on (simply because you waste so much time for no profit). I tried it, and immediately went back to cargo running instead, and for fun I started a combat oriented play as well. Never doing mining again in its current form.

**Comment by tehhowch** [comment]
it's pretty easy to spot iron, neodymium, copper, silver, and lead. Every minable moves clockwise around the star in an eccentric orbit.
One can always piggyback off the AI miners while learning how to spot the minables.

**Comment by Bladewood** [comment]
The regular asteroids have boring textures. You'll know if it's minable when you see it.

**Comment by ghoulavenger** [comment]
And I just said that I couldn't tell the difference for the majority of them. This makes the mining process tedious in real time and not very profitable, despite the minables having high profits. In my view it would be better if the minables were easier to detect but lower value, so that it was comparable to trading. Even if it were slightly better than trading, you would be spending real time rather than in game days, so it should balance itself out. So I have no problem with targettable asteroids, since it solves both of those problems.

Setting the auto-mine and auto-harvest though I can see where the objection is. It makes the game more bottable, which is something that ES is trying not to do. Although you can still technically do it idling in pirate infested systems, since your fleet does automatically engage, with or without holding position.

**Comment by gunqqer** [comment]
If seeing them is the problem it shouldn't be to hard to have a preferance that highlights minables

**Comment by endless-sky** [comment]
You could always just make a mod that paints them neon pink or something.

Really, there's no way to make every part of the game easy for everyone without making it boring for the rest. The mineable asteroids were designed to be possible for most people to tell apart from the unmineable ones, and the fact that they move differently than normal asteroids and are only found in a ring around the star is intended to make asteroid-hunting possible even for someone who is color-blind or has minor vision problems. But some people still won't be able to get the hang of hunting them, the same as some people never really learn to dodge missiles or to chain jobs together in an intelligent fashion.

**Comment by KyroFirehammer** [comment]
That not what I meant but ok. I was talking for ships slower than the planets pracically.....
